homeassistant/components/hangouts/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow to configure Google Hangouts."""
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import callback
+
+from .const import CONF_2FA, CONF_REFRESH_TOKEN
+from .const import DOMAIN as HANGOUTS_DOMAIN
+
+
+@callback
+def configured_hangouts(hass):
+    """Return the configures Google Hangouts Account."""
+    entries = hass.config_entries.async_entries(HANGOUTS_DOMAIN)
+    if entries:
+        return entries[0]
+    return None
+
+
+@config_entries.HANDLERS.register(HANGOUTS_DOMAIN)
+class HangoutsFlowHandler(data_entry_flow.FlowHandler):
+    """Config flow Google Hangouts."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize Google Hangouts config flow."""
+        self._credentials = None
+        self._refresh_token = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        errors = {}
+
+        if configured_hangouts(self.hass) is not None:
+            return self.async_abort(reason="already_configured")
+
+        if user_input is not None:
+            from hangups import get_auth
+            from .hangups_utils import (HangoutsCredentials,
+                                        HangoutsRefreshToken,
+                                        GoogleAuthError, Google2FAError)
+            self._credentials = HangoutsCredentials(user_input[CONF_EMAIL],
+                                                    user_input[CONF_PASSWORD])
+            self._refresh_token = HangoutsRefreshToken(None)
+            try:
+                await self.hass.async_add_executor_job(get_auth,
+                                                       self._credentials,
+                                                       self._refresh_token)
+
+                return await self.async_step_final()
+            except GoogleAuthError as err:
+                if isinstance(err, Google2FAError):
+                    return await self.async_step_2fa()
+                msg = str(err)
+                if msg == 'Unknown verification code input':
+                    errors['base'] = 'invalid_2fa_method'
```

Comment:
Can this happen ? I thought if we get a 2FAerror we already redirect to the 2fa step?

Reply:
jep, this can happen...
if we get this error message, it is not our 2FAError, than the Google Account is secured the 2FA with asking on the phone to accept the new login, this is not supported in hangups.